### DISCUSSION TITLE
SDMA with Sionna

### DISCUSSION THREAD

**Discussion by BaptisteSambon** [NVlabs/sionna#584] [General]
Hi everyone, 

I was wondering how it would be possible to do spatial multiplexing (SDMA) with Sionna to allow users to access the same time-frequency resources but still be able to separate them in the spatial domain. 

I guess one would have to use a multi-user MIMO system and play with the StreamManagement class, but I am not familiar enough with Sionna to see how to actually implement this. 

Many thanks in advance for your help ;-)

**Comment by faycalaa** [comment]
Hi @BaptisteSambon,

I recommend checking out the [tutorials](https://nvlabs.github.io/sionna/tutorials.html). In particular, there's [a tutorial focused on setting up multi-user MIMO simulations](https://nvlabs.github.io/sionna/examples/Realistic_Multiuser_MIMO_Simulations.html).

**Reply by BaptisteSambon** [reply]
Hi @faycalaa, 

Thank you very much for your response and your help! 

I have looked at this tutorial but I have a few misunderstandings about this one and about understanding the results. 

Indeed, I tried to simulate a communication between 1 user and 1 BS in uplink but with the number of antennas equal to 4 and 8 for the UT and the BS respectively. The following parameters were then modified compared to the initial tutorial in the Keras model (at the end of the tutorial): 

        self._num_bs_ant = 8
        self._num_ut = 1
        self._num_ut_ant = 4

      self._ut_array = AntennaArray(
                                       num_rows=1,
                                       num_cols=int(self._num_ut_ant/2),
                                       polarization="single",
                                       polarization_type="V",
                                       antenna_pattern="omni",
                                       carrier_frequency=self._carrier_frequency)

I wanted to understand how the transmitter could perform spatial multiplexing to send multiple independent streams of information in parallel with a classic MIMO structure and how it could benefit from diversity to obtain a diversity gain. With the diversity-multiplexing trade-off, the diversity gain should decreases as the multiplexing gain increases. I tried to observe that in Sionna with the scenario described above by changing the "self._num_streams_per_tx" parameter. As I understand it, this parameter corresponds to the number of parallel streams that can be sent, and therefore to the multiplexing gain. 

However, when I simulate 1 stream per tx or 2 by changing the “self._num_streams_per_tx” parameter to 1 and 2 respectively, I'm surprised to observe that the BER values obtained are lower when the number of streams equals 2 (see image below). This doesn't make sense with my reasoning, since with a single stream of information and multiple antennas at the receiving and transmitting end, the diversity gain should be greater and therefore the BER values obtained lower. 

![image](https://github.com/user-attachments/assets/aedea081-4a56-4e30-be1b-3a2eb4b52a14)

So I am wondering what does the “self._num_streams_per_tx” parameter correspond to ? Does it corresponds to the number of different versions of the same information received at the BS ? But in this case, how could multiplexing be performed ? 

Many thanks in advance for your response and thanks again for your help.
